knimin/lib/data_access.py

```diff
@@ -2415,6 +2415,126 @@ def set_deposited_ebi(self):
+        ValueError
+            If there is error creating the study
+        """
+        with self._con.cursor() as cur:
+            cur.execute('BEGIN')
+            sql = """SELECT study_id, title FROM pm.study WHERE study_id = %s
+                     OR title = %s"""
+            sql_args = [study_id, title or None]
+            qry = self._con.execute_fetchone(sql, sql_args)
+            if qry is not None:
+                raise ValueError('Study ID or title conflicts with exisiting '
+                                 'study %s: %s.' % (qry[0], qry[1]))
+            sql = """INSERT INTO pm.study (study_id, title, alias, notes)
+                     VALUES (%s, %s, %s, %s) RETURNING study_id"""
+            sql_args = [study_id, title or None, alias or None, notes or None]
+            if self._con.execute_fetchone(sql, sql_args) is None:
+                raise ValueError('Creation of study %s failed.' % study_id)
+            cur.execute('COMMIT')
+
+    def edit_study(self, study_id, title=None, alias=None, notes=None):
+        """Edits properties of an existing study
+
+        Parameters
+        ----------
+        study_id : int
+        title : str
+        alias : str
+        notes : str
+
+        Raises
+        ------
+        ValueError
+            If there is error editing the study's properties
+        """
+        with self._con.cursor() as cur:
+            cur.execute('BEGIN')
+            sql = """SELECT * FROM pm.study WHERE study_id = %s LIMIT 1"""
+            sql_args = [study_id]
+            if self._con.execute_fetchone(sql, sql_args) is None:
+                raise ValueError('Study ID %s does not exist.' % study_id)
+            if title:
+                sql = """SELECT study_id FROM pm.study WHERE study_id <> %s
+                         AND title = %s LIMIT 1"""
+                sql_args = [study_id, title]
+                qry = self._con.execute_fetchone(sql, sql_args)
+                if qry is not None:
+                    raise ValueError('Study title "%s" conflicts with another '
+                                     'study: %s.' % (title, qry[0]))
+            sql = """UPDATE pm.study SET title = %s, alias = %s, notes = %s
+                     WHERE study_id = %s RETURNING study_id"""
+            sql_args = [title or None, alias or None, notes or None, study_id]
+            if self._con.execute_fetchone(sql, sql_args) is None:
+                raise ValueError('Editing study %s failed.' % study_id)
+            cur.execute('COMMIT')
+
+    def read_study(self, study_id):
+        """Read properties of an existing study
+
+        Parameters
+        ----------
+        study_id : int
+
+        Returns
+        -------
+        dict
+            {title : str, alias : str, notes : str}
+
+        Raises
+        ------
+        ValueError
+            If there is error reading the study's properties
+        """
+        with self._con.cursor() as cur:
+            cur.execute('BEGIN')
```

Comment:
@wasade Thank you! So operations that do not modify the database do not have to be included in a transaction. Am I getting it right?